app/util/decorators.py

```diff
@@ -27,6 +27,47 @@ def function_with_retries(*args, **kwargs):
     return method_decorator
 
 
+def retry_on_exception_exponential_backoff(exceptions, initial_delay=0.1, max_delay=10):
```

Comment:
Actually that'd only be three arguments, as the max_delay argument would go away if we introduced the total amount of delay.